src/adapters/suggest.js

```diff
@@ -1,258 +1,284 @@
+  </div>
+;
+
+export default class Suggest extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputValue: '',
+      pois: [],
+      favorites: [],
+      categories: [],
+      isOpen: false,
+      selectedItem: null,
+      typedValue: '',
     };
   }
 
-  async preselect(term) {
-    const suggestList = await this.autocomplete.prefetch(term);
-    if (suggestList && suggestList.length > 0) {
-      const firstPoi = suggestList[0];
-      this.onSelect(firstPoi);
-      this.searchInputDomHandler.blur();
-    }
-    return suggestList;
+  async componentDidMount() {
+    const favorites = await PoiStore.getAll();
+    this.setState({
+      favorites,
+    });
   }
 
-  setIdle(idle) {
-    this.searchInputDomHandler.readOnly = idle;
+  componentDidUpdate(prevProps) {
+    // if (this.props.inputValue !== prevProps.inputValue) {
+    //   this.source(this.props.inputValue);
+    // }
   }
 
-  async onSubmit() {
-    if (this.pending) {
-      if (this.bragiPromise) {
-        this.bragiPromise.abort();
-      }
-      const term = this.searchInputDomHandler.value;
-      this.preselect(term);
+  source = debounce(async term => {
+    let promise;
+
+    if (term === '') {
+      // Prerender Favorites on focus in empty field
+      // console.log('should Prerender Favorites');
+      // const favorites = await PoiStore.getAll();
+      // console.log('favorites', favorites);
+      this.setState({
+        pois: [],
+      });
     } else {
-      if (this.suggestList && this.suggestList.length > 0 &&
-          this.searchInputDomHandler.value &&
-          this.searchInputDomHandler.value.length > 0) {
-        this.onSelect(this.suggestList[0]);
-        this.searchInputDomHandler.blur();
-      }
-    }
-  }
-
-  destroy() {
-    this.autocomplete.destroy();
-  }
-
-  preRender(items) {
-    this.autocomplete.preRender(items);
-  }
-
-  prefixesRender() {
-    return this.prefixes.map(prefix => prefix.render());
-  }
-
-  remotesRender(pois) {
-    return pois.map(poi => this.renderItem(poi)).join('');
-  }
+      promise = () => new Promise(async (resolve, reject) => {
+        const focus = {};
+        const mapZoom = window.map && window.map.mb && window.map.mb.getZoom();
+        if (SUGGEST_USE_FOCUS && mapZoom >= SUGGEST_FOCUS_MIN_ZOOM) {
+          const center = window.map.mb.getCenter();
+          focus.lat = center.lat;
+          focus.lon = center.lng;
+          focus.zoom = mapZoom;
+        }
+        this.historyPromise = PoiStore.get(term);
+        this.bragiPromise = BragiPoi.get(term, focus);
+        this.categoryPromise = this.props.withCategories ?
+          CategoryService.getMatchingCategories(term) : null;
+
+        try {
+          const [bragiResponse, storeResponse, categoryResponse] = await Promise.all([
+            this.bragiPromise, this.historyPromise, this.categoryPromise,
+          ]);
+
+          if (!bragiResponse) {
+            return resolve(null);
```

Comment:
It looks like it will ignore suggestions from store and category. Is it what we want?